File: code/backend/java/core/src/main/java/games/paths/core/model/story/CharacterTemplateInfo.java
package games.paths.core.model.story;

/**
 * CharacterTemplateInfo - Domain model for a character template summary.
 * Used within StoryDetail to describe available character templates for a story.
 * Immutable record.
 */
public record CharacterTemplateInfo(
        String uuid,
        String name,
        String description,
        int lifeMax,
        int energyMax,
        int sadMax,
        int dexterityStart,
        int intelligenceStart,
        int constitutionStart,
        Integer idCard,
        CardInfo card,
        Integer idClassPermitted,
        Integer idClassProhibited) {
}

File: code/backend/python/app/core/models/story/character_template_info.py
from dataclasses import dataclass
from app.core.models.story.card_info import CardInfo

@dataclass
class CharacterTemplateInfo:
    uuid: str
    name: str | None = None
    description: str | None = None
    lifeMax: int = 0
    energyMax: int = 0
    sadMax: int = 0
    dexterityStart: int = 0
    intelligenceStart: int = 0
    constitutionStart: int = 0
    idCard: int | None = None
    card: CardInfo | None = None
